common/perf-test-core/src/main/java/com/azure/perf/test/core/HttpPerfTestBase.java

```diff
@@ -0,0 +1,243 @@
+     * it can be configured manually using the "httpClient" and "policies" fields.
+     * @param clientBuilder The client builder.
+     * @throws IllegalStateException If reflective access to get httpClient or addPolicy methods fail.
+     */
+    protected void configureClientBuilder(Object clientBuilder) {
+        if (httpClient != null || policies != null) {
+            Class<?> clientBuilderClass = clientBuilder.getClass();
+
+            try {
+                if (httpClient != null) {
+                    Method httpClientMethod = clientBuilderClass.getMethod("httpClient", HttpClient.class);
+                    httpClientMethod.invoke(clientBuilder, httpClient);
+                }
+
+                if (policies != null) {
+                    Method addPolicyMethod = clientBuilderClass.getMethod("addPolicy", HttpPipelinePolicy.class);
+                    for (HttpPipelinePolicy policy : policies) {
+                        addPolicyMethod.invoke(clientBuilder, policy);
+                    }
+                }
+            } catch (ReflectiveOperationException e) {
+                throw new IllegalStateException(e);
+            }
+        }
+    }
+
+    @Override
+    public void runAll(long endNanoTime) {
+        completedOperations = 0;
+        lastCompletionNanoTime = 0;
+        long startNanoTime = System.nanoTime();
+        while (System.nanoTime() < endNanoTime) {
+            completedOperations += runTest();
+            lastCompletionNanoTime = System.nanoTime() - startNanoTime;
+        }
+    }
+
+    @Override
+    public Mono<Void> runAllAsync(long endNanoTime) {
+        completedOperations = 0;
+        lastCompletionNanoTime = 0;
+        long startNanoTime = System.nanoTime();
+
+        return Flux.just(1)
+            .repeat()
+            .flatMap(i -> runTestAsync(), 1)
+            .doOnNext(v -> {
+                completedOperations +=v;
+                lastCompletionNanoTime = System.nanoTime() - startNanoTime;
+            })
+            .takeWhile(i -> System.nanoTime() < endNanoTime)
+            .then();
+    }
+
+    /**
+     * Indicates how many operations were completed in a single run of the test.
+     * Good to be used for batch operations.
+     *
+     * @return the number of successful operations completed.
+     */
+    abstract int runTest();
+
+    /**
+     * Indicates how many operations were completed in a single run of the async test.
+     * Good to be used for batch operations.
+     *
+     * @return the number of successful operations completed.
+     */
+    abstract Mono<Integer> runTestAsync();
+
+    /**
+     * Stops playback tests.
+     * @return An empty {@link Mono}.
+     */
+    @Override
+    public Mono<Void> stopPlaybackAsync() {
+        return recordPlaybackHttpClient
+            .headers(h -> {
+                h.set("x-recording-id", recordingId);
+                h.set("x-purge-inmemory-recording", Boolean.toString(true));
+            })
```

Comment:
Adding some documentation here would be good. What do these headers do and how are they used?

Reply:
added